pyvista/core/dataobject.py

```diff
@@ -156,6 +157,17 @@ def save(
 
         """
 
+        def _warn_multiblock_nested_field_data(mesh: pyvista.MultiBlock) -> None:
+            for index, name, nested_multiblock in mesh.recursive_iterator(
+                'all', node_type='parent'
+            ):
+                index_fmt = ''.join([f'[{ind}]' for ind in index])
+                if len(nested_multiblock.field_data.keys()) > 0:
+                    warnings.warn(
+                        f"Nested MultiBlock at index {index_fmt} with name '{name}' has field data which will not be saved.\n"
+                        'See https://gitlab.kitware.com/vtk/vtk/-/issues/19414'
```

Comment:
Extend the message to suggest moving field data before saving. Depends on #7251 .

```suggestion
                        f"Nested MultiBlock at index {index_fmt} with name '{name}' has field data which will not be saved.\n"
                        'See https://gitlab.kitware.com/vtk/vtk/-/issues/19414 \n'
                        'Use `move_nested_field_data_to_root` to store the field data with the root MultiBlock before saving.'
```